scripts/patrol/patrol.py

```diff
@@ -352,11 +352,53 @@ def _check_constraints(self, patrol: PatrolEvent) -> bool:
             
         return True
 
+    def _get_cats_from_tag(self, cat1, cat2):
+
+        app_abbr = ["app1", "app2", "app3", "app4", "app5", "app6"]
+        o_c_abbr = ["o_c1", "o_c2", "o_c3", "o_c4"]
+        other_cats = [i for i in self.patrol_cats if i not in [self.patrol_leader, self.patrol_random_cat, self.patrol_apprentices]]
+
+        for inter_cat in self.patrol_cats:
+            if inter_cat.ID == self.patrol_leader.ID:
+                if cat1 == "p_l":
+                    cat1 = inter_cat
+                elif cat2 == "p_l":
+                    cat2 = inter_cat
+            elif inter_cat.ID == self.patrol_random_cat.ID:
+                if cat1 == "r_c":
+                    cat1 = inter_cat
+                elif cat2 == "r_c":
+                    cat2 = inter_cat
+           
+        if (len(self.patrol_apprentices) == 0) and ((cat1 in app_abbr) or (cat2 in app_abbr)):
+            cat1 = None
+        for i, abbr in enumerate(app_abbr):
+            if abbr in [cat1, cat2]:
+                if len(self.patrol_apprentices) > i:
+                    if cat1 == abbr:
+                        cat1 = self.patrol_apprentices[i]
+                    else:
+                        cat2 = self.patrol_apprentices[i]
+            
+        if (len(other_cats) == 0) and ((cat1 in o_c_abbr) or (cat2 in o_c_abbr)):
+            cat1 = None
+        i for i, abbr in enumerate(o_c_abbr):
```

Comment:
```suggestion
        for i, abbr in enumerate(o_c_abbr):
```
This is what's throwing that test error :)